File: templates/go/file.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

func readInput() []string {
	lines := []string{}
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}
	return lines
}

func part1(_lines []string) int {
	return 0
}

func part2(_lines []string) int {
	return 0
}

func main() {
	s := time.Now()
	lines := readInput()
	s1 := time.Now()
	fmt.Println(part1(lines))
	e1 := time.Since(s1)
	s2 := time.Now()
	fmt.Println(part2(lines))
	e2 := time.Since(s2)
	e := time.Since(s)
	fmt.Printf("Time for part1: %s, part2: %s, total: %s\n", e1, e2, e)
}

File: templates/rb/file.rb
# frozen_string_literal: true

def part1(lines) = 0
def part2(lines) = 0
def tdiff(s,e) = "#{(e - s) * 1000}ms"

lines = ARGF.readlines
s = Process.clock_gettime(Process::CLOCK_MONOTONIC)
p part1 lines
e1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
p part2 lines
e2 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
puts "Time taken part1: #{tdiff(s,e1)}, part2: #{tdiff(e1,e2)}, total: #{tdiff(s,e2)}"
